validation/gpo-params_test.go

```diff
@@ -0,0 +1,109 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func TestGasPriceOracleParams(t *testing.T) {
+	isExcluded := map[uint64]bool{
+		291:          true, // mainnet/orderly                 (incorrect scalar parameter)
+		888:          true, // goerli-dev-0/op-labs-chaosnet-0 (no public endpoint)
+		957:          true, // mainnet/lyra                    (incorrect scalar parameter)
+		997:          true, // goerli-dev-0/op-labs-devnet-0   (no public endpoint)
+		58008:        true, // sepolia/pgn                     (incorrect overhead parameter)
+		84532:        true, // sepolia/base                    (incorrect overhead parameter)
+		11155421:     true, // sepolia-dev-0/oplabs-devnet-0   (no public endpoint)
+		11763071:     true, // goerli-dev-0/base-devnet-0      (no public endpoint)
+		11763072:     true, // sepolia-dev-0/base-devnet-0     (no public endpoint)
+		129831238013: true, // goerli-dev-0/conduit-devnet-0   (no ground truth)
+	}
+
+	checkResourceConfig := func(t *testing.T, chain *ChainConfig) {
+		rpcEndpoint := chain.PublicRPC
+		require.NotEmpty(t, rpcEndpoint, "no public endpoint for chain")
+
+		client, err := ethclient.Dial(rpcEndpoint)
+		require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+		contractAddress := predeploys.GasPriceOracleAddr
+
+		var desiredParams GasPriceOracleParams
+		switch chain.Superchain {
+		case "mainnet":
+			desiredParams = OPMainnetGasPriceOracleParams
+		case "goerli":
+			desiredParams = OPGoerliGasPriceOracleParams
+		case "sepolia":
+			desiredParams = OPSepoliaGasPriceOracleParams
+		case "goerli-dev-0":
+			t.Fatalf("no ground truth for superchain %s", chain.Superchain)
+		case "sepolia-dev-0":
+			t.Fatalf("no ground truth for superchain %s", chain.Superchain)
+		default:
+			t.Fatalf("superchain not recognized: %s", chain.Superchain)
+		}
+
+		actualParams, err := getGasPriceOracleParamsWithRetries(context.Background(), contractAddress, client)
+		require.NoErrorf(t, err, "RPC endpoint %s: %s", rpcEndpoint)
+
+		require.Condition(t, func() bool { return (actualParams.Decimals.Cmp(desiredParams.Decimals) == 0) },
+			"incorrect decimals parameter: got %d, wanted %d", actualParams.Decimals, desiredParams.Decimals)
+		require.Condition(t, func() bool { return (actualParams.Overhead.Cmp(desiredParams.Overhead) == 0) },
```

Comment:
More readable IMO to use `require.Equal(t, x.Cmp(y), 0)`. You may even be able to use `require.EqualValues`.

Reply:
[4da6419](https://github.com/ethereum-optimism/superchain-registry/pull/80/commits/4da6419cc34863a8e8d546dd5a8b5dc80d62956c)